controller/tests/array_action/svc/array_mediator_svc_test.py

```diff
@@ -341,32 +411,42 @@ def test_delete_snapshot_no_fcmap_id_raise_snapshot_not_found(self):
         with self.assertRaises(array_errors.ObjectNotFoundError):
             self.svc.delete_snapshot("test_snap")
 
-    def test_delete_snapshot_call_rmvolume(self):
-        self._prepare_mocks_for_get_snapshot()
-
-        self.svc.delete_snapshot("test_snap")
-
-        self.svc.client.svctask.rmvolume.assert_called_once_with(vdisk_id="test_snap")
-
     def test_delete_snapshot_call_rmfcmap(self):
-        self._prepare_mocks_for_get_snapshot()
-
+        self._prepare_mocks_for_delete_snapshot()
+        fcmaps_as_target = self.fcmaps
+        self.svc.client.svcinfo.lsfcmap.side_effect = [Mock(as_list=fcmaps_as_target), Mock(as_list=[])]
         self.svc.delete_snapshot("test_snap")
 
         self.svc.client.svctask.rmfcmap.assert_called_once_with(object_id="test_fc_id", force=True)
 
     @patch("controller.array_action.array_mediator_svc.is_warning_message")
     def test_delete_snapshot_rmvolume_error(self, mock_warning):
-        self._prepare_mocks_for_get_snapshot()
+        self._prepare_mocks_for_delete_snapshot()
         mock_warning.return_value = False
         self.svc.client.svctask.rmvolume.side_effect = [
             CLIFailureError("Failed")]
 
         with self.assertRaises(CLIFailureError):
             self.svc.delete_snapshot("test_snap")
 
+    @patch("controller.array_action.array_mediator_svc.is_warning_message")
+    def test_snapshot_still_copy_fcmaps_not_removed(self, mock_warning):
+        cli_volume = self._prepare_mocks_for_delete_snapshot()
+        cli_volume.FC_id = 'many'
+        mock_warning.return_value = False
+        fcmaps_as_source = self.fcmaps
+        fcmaps_as_source[0].status = "not good"
+        fcmaps_as_source[0].source_vdisk_name = "test_snap"
+        self.svc.client.svcinfo.lsfcmap.side_effect = [Mock(as_list=fcmaps_as_source), Mock(as_list=fcmaps_as_source)]
+        with self.assertRaises(array_errors.ObjectIsStillInUseError):
+            self.svc.delete_snapshot("test_snap")
+
     def test_delete_snapshot_success(self):
+        cli_volume = self._prepare_mocks_for_delete_snapshot()
+        cli_volume.FC_id = 'many'
```

Comment:
should this be inside `_prepare_mocks_for_delete_snapshot`?

Reply:
done